internal/experimental/registry/client.go

```diff
@@ -117,7 +119,17 @@ func (c *Client) Logout(hostname string) error {
 }
 
 // PushChart uploads a chart to a registry
-func (c *Client) PushChart(ref *Reference) error {
+func (c *Client) PushChart(ref *Reference, insecure bool, plainHTTP bool) error {
+	if insecure || plainHTTP {
```

Comment:
The if condition maybe not necessary? Because `c.newResolver` have check `insecure` true or false.

Reply:
In line 79: `resolver, err := client.newResolver(false, false)` set with default resolver with `insecure=false` and `plainHTTP=false`. So i just want to replace the default resolver when `insecure = true` or `plainHTTP = true`, and still use the default resolver in other cases.